### ISSUE TITLE
[BUG] False positive mypy error when using telegram.ext

### ISSUE THREAD

**Issue by revolter** [python-telegram-bot/python-telegram-bot#2580]
### Steps to reproduce

1. Install `mypy`
2. Write `import telegram.ext` and `var = telegram.ext.CallbackContext()`
3. Run `mypy`

### Expected behaviour

Only the error
```
Too few arguments for "CallbackContext"  [call-arg] (3:7)
```
should be thrown.

I'm pretty sure this used to be ok in a previous version.

### Actual behaviour

The errors
```
Too few arguments for "CallbackContext"  [call-arg] (3:7)
Need type annotation for 'var'  [var-annotated] (3:7)
```
are thrown.

### Configuration
**Operating System:**

macOS 11.4 (20F71)

**Version of Python, python-telegram-bot & dependencies:**

```
python-telegram-bot 13.7
Bot API 5.3
certifi 2020.12.05
Python 3.9.5 (default, May  4 2021, 03:36:27)  [Clang 12.0.0 (clang-1200.0.32.29)]
```

### Logs
Insert logs here (if necessary)

**Comment by Bibo-Joshi** [comment]
Hi, I can reproduce. The problem is that the `__init__` of `CallbackContext` isn't annotated thoroughly enough. We'd have to do something like here:

https://github.com/python-telegram-bot/python-telegram-bot/blob/bcec6f03cb1797d31039926f0c8c0f2306dcdf5a/telegram/ext/dispatcher.py#L210-L221

i.e. annotate `self` with the same typevars that the `dispatcher` argument is annotated with (though I don't think overloads will be necessary). Maybe you want to PR for this?

However, I'd like to point out that there should normally be no need to instantiate `CallbackContext` yourself …

**Comment by revolter** [comment]
> Maybe you want to PR for this?

If I replace

```diff
-    def __init__(self, dispatcher: 'Dispatcher'):
+    def __init__(self, dispatcher: 'Dispatcher[CCT, UD, CD, BD]'):
```

the `flake8` and `mypy-ptb` pre-commit hooks fail with the errors

```
telegram/ext/callbackcontext.py:108:36: F821 undefined name 'CCT'
```

and

```
telegram/ext/callbackcontext.py:108: error: Name 'CCT' is not defined  [name-defined]
```

---

If I replace

```diff
-from telegram.ext.utils.types import UD, CD, BD
+from telegram.ext.utils.types import CCT, UD, CD, BD
```

the `pylint` pre-commit hook fails with the error

```
telegram/ext/callbackcontext.py:38:0: W0611: Unused CCT imported from telegram.ext.utils.types (unused-import)
```

So they're contradicting themselves :(

**Comment by revolter** [comment]
> I'd like to point out that there should normally be no need to instantiate `CallbackContext` yourself …

I know, it's only for debug purposes/

**Comment by Bibo-Joshi** [comment]
Regarding pre-commit querks: If you put the type hints as string (`'Dispatcher[CCT, UD, CD, BD]'` instead of `Dispatcher[CCT, UD, CD, BD]`) - which is perfectly fine to avoid cyclical imports! - the typevars are not used at runtime, so you can import under the `if TYPE_CHECKING:`  clause. If that's not enough to make pylint happy, just add a `  # pylint: disable=ERRORCODE` on the corresponding line - pylint is clever but not infallible ;)

Annotation the `dispatcher` argument alone won't suffice, though, you'll have to annotate `self` as well :)

> I know, it's only for debug purposes/
fair enough :)

**Comment by Bibo-Joshi** [comment]
closed by #2587
